ruby-package/lib/base_thrift_client.rb

```diff
@@ -32,6 +33,20 @@ def get_executor
         @executor
       end
 
+
+      def with_retries(&f)
+        (0..@max_retries).each do |retry_count|
+          begin
+            return f.call(get_executor)
+          rescue Thrift::TransportException, IOError
+            if retry_count == @max_retries then
+              raise # re-throw exception back to user
+            else
+              puts "Attempting to reconnect to #{@server}:#{@port}..."
```

Comment:
Maybe we can also include the caller method name in the log message? http://stackoverflow.com/questions/5100299/how-to-get-the-name-of-the-calling-method